### PR TITLE
[DTM] SOFT-4083 [43/51]: extract normalize.js's per-kind logic into kinds/ modules

### PR DESCRIPTION
**Pull request by pauloiankoski**
Splits `token-indicators/normalize.js`'s per-kind comparison logic out into dedicated `kinds/` modules (dimension, color, border, text), leaving `normalize.js` as a thin dispatch wrapper.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling and matching of border widths, styles, and colors, including per-side values and empty states.
  * Improved color normalization for palette references and literal color values.
  * Improved responsive dimension handling, preset inheritance, units, and per-corner values.
  * Standardized text value normalization by trimming whitespace and handling empty values.

* **Tests**
  * Added comprehensive coverage for border, color, dimension, and text normalization and preset matching.

### CODE REVIEW

**Comment by pauloiankoski** [comment]
@coderabbitai review

File: src/extension/token-indicators/normalize.js [hunk 1]
```diff
@@ -358,140 +96,7 @@ function borderSource(value) {
...
-		return BORDER_SIDES.every(
-			(side) => normalizeColor((source[side] || [])[index]) === normalizeColor(presetValue)
-		);
-	}
-
-	return BORDER_SIDES.every(
-		(side) => normalizeText((source[side] || [])[index] || 'none') === normalizeText(presetValue)
-	);
+	return KINDS[kind].isEmpty(value);
 }
```
[the remark this answers is not in the document]
Comment (pauloiankoski) [reply src/extension/token-indicators/normalize.js:116]: Fixed — `isEmptyValue`/`matchesPreset` now fall back to the `text` handler for a kind not in `KINDS`, with tests covering the fallback.

**Review by d4mation — APPROVED** [review]
